service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/CloneControlledAzureDatabaseResourceFlight.java

```diff
@@ -1,12 +1,85 @@
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import java.util.List;
+import java.util.Optional;
 
-public class CloneControlledAzureDatabaseResourceFlight extends Flight {
+public class CloneControlledAzureDatabaseResourceFlight extends CloneControlledAzureResourceFlight {
 
   public CloneControlledAzureDatabaseResourceFlight(
       FlightMap inputParameters, Object applicationContext) {
     super(inputParameters, applicationContext);
   }
+
+  @Override
+  protected List<StepRetryRulePair> copyDefinition(
+      FlightBeanBag flightBeanBag, FlightMap inputParameters) {
+    var sourceDatabase =
+        FlightUtils.getRequired(
+            inputParameters,
+            WorkspaceFlightMapKeys.ResourceKeys.RESOURCE,
+            ControlledAzureDatabaseResource.class);
+    var userRequest =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+
+    var cloningInstructions =
+        Optional.ofNullable(
+                inputParameters.get(
+                    WorkspaceFlightMapKeys.ResourceKeys.CLONING_INSTRUCTIONS,
+                    CloningInstructions.class))
+            .orElse(sourceDatabase.getCloningInstructions());
+
+    return List.of(
+        new StepRetryRulePair(
+            new CopyControlledAzureDatabaseDefinitionStep(
+                flightBeanBag.getSamService(),
+                userRequest,
+                sourceDatabase,
+                flightBeanBag.getControlledResourceService(),
+                cloningInstructions,
+                flightBeanBag.getWsmResourceService()),
+            RetryRuleNone.getRetryRuleNone())); // CreateDatabase already retries
+  }
+
+  @Override
+  protected List<StepRetryRulePair> copyResource(
+      FlightBeanBag flightBeanBag, FlightMap inputParameters) {
+
+    var sourceDatabase =
+        FlightUtils.getRequired(
+            inputParameters,
+            WorkspaceFlightMapKeys.ResourceKeys.RESOURCE,
+            ControlledAzureDatabaseResource.class);
+
+    return List.of(
+        new StepRetryRulePair(
+            new DumpAzureDatabaseStep(
+                sourceDatabase,
+                flightBeanBag.getLandingZoneApiDispatch(),
+                flightBeanBag.getSamService(),
+                flightBeanBag.getWorkspaceService(),
+                flightBeanBag.getAzureStorageAccessService(),
+                flightBeanBag.getAzureDatabaseUtilsRunner(),
+                flightBeanBag.getWsmResourceService()),
+            RetryRules.cloud()),
+        new StepRetryRulePair(
+            new RestoreAzureDatabaseStep(
+                flightBeanBag.getLandingZoneApiDispatch(),
+                flightBeanBag.getSamService(),
+                flightBeanBag.getWorkspaceService(),
+                flightBeanBag.getAzureStorageAccessService(),
+                flightBeanBag.getAzureDatabaseUtilsRunner()),
+            RetryRules.cloud()));
```

Comment:
do we want to delete the dump file?

Reply:
From my understanding this will be deleted by followup ticket WM-2372 which will delete the container the dumpfile is in